streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedTasks.java

```diff
@@ -279,7 +261,7 @@ boolean maybeResumeSuspendedTask(final TaskId taskId, final Set<TopicPartition>
         return false;
     }
 
-    private void addToRestoring(final Task task) {
+    private void addToRestoring(final T task) {
```

Comment:
As task restoration is only done for active tasks should we move all methods and fields to do with restoring to the `AssignedStreamsTask` class ?

Reply:
Hmmm... even if `addToRestoring` is only called for active tasks, we use it within `initializeNewTasks` that is used for both (active and standby). Not sure if we could do this in an elegant way...

Reply:
Looked into this part as well. I think extracting `initializeNewTasks` for active tasks and standby tasks is a bit overkill than just letting `StandbyTask.initialize()` return true.